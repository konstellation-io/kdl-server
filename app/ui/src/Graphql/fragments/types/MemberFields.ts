/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AccessLevel } from "./../../types/globalTypes";

// ====================================================
// GraphQL fragment: MemberFields
// ====================================================

export interface MemberFields_user {
  __typename: "User";
  id: string;
  email: string;
  lastActivity: string | null;
}

export interface MemberFields {
  __typename: "Member";
  user: MemberFields_user;
  accessLevel: AccessLevel;
  addedDate: string;
}
