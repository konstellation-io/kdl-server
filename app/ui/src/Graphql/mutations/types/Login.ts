/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

// ====================================================
// GraphQL mutation operation: Login
// ====================================================

export interface Login_login {
  __typename: "User";
  id: string;
}

export interface Login {
  login: Login_login;
}
