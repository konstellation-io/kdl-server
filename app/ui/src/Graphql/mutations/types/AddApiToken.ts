/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { ApiTokenInput } from "./../../types/globalTypes";

// ====================================================
// GraphQL mutation operation: AddApiToken
// ====================================================

export interface AddApiToken_addApiToken {
  __typename: "ApiToken";
  id: string;
  name: string;
  creationDate: string;
  lastUsedDate: string;
  token: string;
}

export interface AddApiToken {
  addApiToken: AddApiToken_addApiToken | null;
}

export interface AddApiTokenVariables {
  input: ApiTokenInput;
}
