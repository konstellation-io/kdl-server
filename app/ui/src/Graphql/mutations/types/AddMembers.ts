/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { AddMembersInput, AccessLevel } from '../../types/globalTypes';

// ====================================================
// GraphQL mutation operation: AddMembers
// ====================================================

export interface AddMembers_addMembers_members_user {
  __typename: 'User';
  id: string;
  email: string;
  lastActivity: string | null;
}

export interface AddMembers_addMembers_members {
  __typename: 'Member';
  user: AddMembers_addMembers_members_user;
  accessLevel: AccessLevel;
  addedDate: string;
}

export interface AddMembers_addMembers {
  __typename: 'Project';
  id: string;
  members: AddMembers_addMembers_members[];
}

export interface AddMembers {
  addMembers: AddMembers_addMembers;
}

export interface AddMembersVariables {
  input: AddMembersInput;
}
