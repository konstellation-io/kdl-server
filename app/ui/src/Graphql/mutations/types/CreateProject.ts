/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { CreateProjectInput } from "./../../types/globalTypes";

// ====================================================
// GraphQL mutation operation: CreateProject
// ====================================================

export interface CreateProject_createProject_repository {
  __typename: "Repository";
  url: string;
  error: string | null;
}

export interface CreateProject_createProject {
  __typename: "Project";
  id: string;
  name: string;
  description: string;
  favorite: boolean;
  creationDate: string;
  lastActivationDate: string;
  repository: CreateProject_createProject_repository | null;
  needAccess: boolean;
  archived: boolean;
  error: string | null;
}

export interface CreateProject {
  createProject: CreateProject_createProject;
}

export interface CreateProjectVariables {
  input: CreateProjectInput;
}
