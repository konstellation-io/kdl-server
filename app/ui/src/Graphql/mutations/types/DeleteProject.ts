/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { DeleteProjectInput } from "./../../types/globalTypes";

// ====================================================
// GraphQL mutation operation: DeleteProject
// ====================================================

export interface DeleteProject_deleteProject {
  __typename: "Project";
  id: string;
}

export interface DeleteProject {
  deleteProject: DeleteProject_deleteProject | null;
}

export interface DeleteProjectVariables {
  input: DeleteProjectInput;
}
