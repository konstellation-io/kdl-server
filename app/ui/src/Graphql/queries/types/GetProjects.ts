/* tslint:disable */
/* eslint-disable */
// @generated
// This file was automatically generated and should not be edited.

import { RepositoryType, AccessLevel } from './../../types/globalTypes';

// ====================================================
// GraphQL query operation: GetProjects
// ====================================================

export interface GetProjects_projects_repository_external {
  __typename: 'ExternalRepository';
  username: string;
}

export interface GetProjects_projects_repository {
  __typename: 'Repository';
  type: RepositoryType;
  url: string;
  error: string | null;
  external: GetProjects_projects_repository_external | null;
}

export interface GetProjects_projects_toolUrls {
  __typename: 'ToolUrls';
  drone: string;
  gitea: string;
  jupyter: string;
  minio: string;
  mlflow: string;
  vscode: string;
}

export interface GetProjects_projects_members_user {
  __typename: 'User';
  id: string;
  email: string;
  lastActivity: string | null;
}

export interface GetProjects_projects_members {
  __typename: 'Member';
  user: GetProjects_projects_members_user;
  accessLevel: AccessLevel;
  addedDate: string;
}

export interface GetProjects_projects {
  __typename: 'Project';
  id: string;
  name: string;
  description: string;
  favorite: boolean;
  creationDate: string;
  lastActivationDate: string;
  repository: GetProjects_projects_repository | null;
  archived: boolean;
  error: string | null;
  toolUrls: GetProjects_projects_toolUrls;
  members: GetProjects_projects_members[];
}

export interface GetProjects {
  projects: GetProjects_projects[];
}
