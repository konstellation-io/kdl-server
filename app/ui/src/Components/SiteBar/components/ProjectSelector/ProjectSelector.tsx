import React, { FC } from 'react';
import styles from './ProjectSelector.module.scss';
import { GetProjects_projects } from 'Graphql/queries/types/GetProjects';
import { NavLink } from 'react-router-dom';
import ROUTE, { buildRoute } from 'Constants/routes';
import { Button } from 'kwc';
import IconAdd from '@material-ui/icons/Add';
import { BottomComponentProps } from '../Breadcrumbs/components/Crumb/Crumb';

type Props = {
  options: GetProjects_projects[];
};
const ProjectSelector: FC<Props & BottomComponentProps> = ({
  options,
  closeComponent = () => {},
}) => (
  <div className={styles.container}>
    <ul>
      {options.map(({ id, name }: GetProjects_projects) => (
        <NavLink
          to={buildRoute(ROUTE.PROJECT, id)}
          key={id}
          onClick={closeComponent}
          activeClassName={styles.selectedProject}
          className={styles.project}
        >
          <li className={styles.projectName}>{name}</li>
        </NavLink>
      ))}
    </ul>
    <Button
      Icon={IconAdd}
      label="New project"
      className={styles.addProjectButton}
      to={ROUTE.NEW_PROJECT}
    />
  </div>
);
export default ProjectSelector;
