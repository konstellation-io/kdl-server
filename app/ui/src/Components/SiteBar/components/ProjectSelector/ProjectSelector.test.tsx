import { GetProjects_projects } from 'Graphql/queries/types/GetProjects';
import { NavLink } from 'react-router-dom';
import ProjectSelector from './ProjectSelector';
import React from 'react';
import { shallow } from 'enzyme';

const project: GetProjects_projects = {
  id: 'foo',
  name: 'bar',
  __typename: 'baz',
  creationDate: 'foofoo',
  description: 'foobar',
  error: 'foobaz',
  favorite: true,
  lastActivationDate: 'barfoo',
  repository: null,
  state: 'STARTED',
};
const projects: GetProjects_projects[] = [project];

const props = {
  options: projects,
  serverId: 'bar',
  closeComponent: jest.fn(),
};

let component;
beforeEach(() => {
  component = shallow(<ProjectSelector {...props} />);
});

describe('ProjectSelector component', () => {
  it('should render without crashing', function () {
    expect(component).toMatchSnapshot();
  });
});

describe('behavior', () => {
  it('should call closeComponent function when click on a link', () => {
    // Arrange.
    const navLink = component.find(NavLink);

    // Act.
    navLink.simulate('click');

    // Assert.
    expect(props.closeComponent).toHaveBeenCalled();
  });
});
