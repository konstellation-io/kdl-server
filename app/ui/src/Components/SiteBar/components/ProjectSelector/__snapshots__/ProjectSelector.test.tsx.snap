// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProjectSelector component should render without crashing 1`] = `
<div
  className="container"
>
  <ul>
    <NavLink
      activeClassName="selectedProject"
      className="project"
      key="foo"
      onClick={[MockFunction]}
      to="/projects/foo"
    >
      <li
        className="projectName"
      >
        bar
      </li>
    </NavLink>
  </ul>
  <T
    Icon={
      Object {
        "$$typeof": Symbol(react.memo),
        "compare": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "render": [Function],
        },
      }
    }
    className="addProjectButton"
    label="New project"
    to="/new-project"
  />
</div>
`;
